components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -440,24 +493,38 @@ void AtmosphereOutput::run (const std::string& filename, const bool is_write_ste
 long long AtmosphereOutput::
 res_dep_memory_footprint () const {
   long long rdmf = 0;
-  if (m_io_field_mgr != m_sim_field_mgr) {
-    // The IO is done on a different grid. The FM stored here is
-    // not shared with anyone else, so we can safely add its footprint
-    for (const auto& it : *m_io_field_mgr) {
-      const auto& fap = it.second->get_header().get_alloc_properties();
-      if (fap.is_subfield()) {
-        continue;
+  const auto sim_field_mgr = get_field_manager("sim");
+
+  // Cycle through all unique field managers in this output,
+  // first make a copy so we can grab just unique pointers.
+  std::vector<std::shared_ptr<const fm_type>> unique_fms;
```

Comment:
This sorting is overly complicated, IMHO. All you need is to track which FM you already parsed. You could do:
```
std::set<std::string> grids;
for (auto fm : m_field_mgrs) {
  const auto& gn = fm->get_grid()->name();
  if (grids.count(gn)>0) {
    continue; // grid manager already parsed
  }
  grids.insert (gn);
  // do stuff on this FM.
}
```